dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+from typing import TypeVar
+from dLux.utils import (get_positions_vector, get_pixel_positions)
+from abc import ABC, abstractmethod 
+import equinox as eqx
+import jax.numpy as np
+import jax 
+import functools
+
+jax.config.update("jax_enable_x64", True)
```

Comment:
We don't want to do this by default, this should be done on a script by script basis